### DISCUSSION TITLE
javascript hook

### DISCUSSION THREAD

**Discussion by nafea107** [livewire/livewire#7332] [3. Feature ideas]
i need javascript hook which i can call on mounted,after updating component and before unmounting

**Comment by Nanochan217** [comment]
🤔 I think what you are looking for may be this, also take into account the component object in the same page:
https://livewire.laravel.com/docs/javascript#dom-morph-hooks

**Comment by nafea107** [comment]
I am using wire:change for a field.
i tried to use component:init for on mounted and it works.
but there are no hooks for befor unmount, before update or after update.
i tried to use morph:update but this hook fallback function called 26 times for one change on the field and one request for the server. I think there is some limitations while using Javascript libraries as it requires DOM element and it works only when initializing the page but after updating DOM i can't initialize the element again as i dont have hook or access to the new DOM. Can you help me if i missunderstood any function or hook

**Comment by nafea107** [comment]
I am using wire:change for a field.
i tried to use component:init for on mounted and it works.
but there are no hooks for befor unmount, before update or after update.
i tried to use morph:update but this hook fallback function called 26 times for one change on the field and one request for the server. I think there is some limitations while using Javascript libraries as it requires DOM element and it works only when initializing the page but after updating DOM i can't initialize the element again as i dont have hook or access to the new DOM. Can you help me if i missunderstood any function or hook

**Comment by Nanochan217** [comment]
Let me gather some info:
Are you using Livewire 3?
What do you need to do specifically?

Reading your response, I think that you want to access to the HTML rendered by a external library in the DOM, right?
If so, whenever Livewire makes a change, the DOM updates as you know, this conflicts with external libraries, because of this, you must use something like "#[Renderless]" attribute to prevent to updating the DOM after making a change or if you need to change the DOM, you can wrap the library rendered HTML or something else in a div with "wire:ignore" HTML directive, this will indicate to Livewire to not refresh this section.

Excuse me if I didn't understand something 😅
